ROOTFS/etc/waggle/sanity/fatal/core_sdcard.test

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+pf="Recovery SDCard test"
+echo "$pf: Begin"
+
+# run fsck on our sd card
+fsck -n -f /dev/mmcblk1p1
+
+if [ "$?" != "0" ];
+then
+        echo "$pf: /dev/mmcblk1p1 failed fsck FAIL"
+        exit 1
+fi
+
+fsck -n -f /dev/mmcblk1p2
+
+if [ "$?" != "0" ];
+then
+        echo "$pf: /dev/mmcblk1p2 failed fsck FAIL"
+        exit 2
+fi
+
+
+mkdir /tmp/core_sdcard_test
+
+mount -r /dev/mmcblk1p1  /tmp/core_sdcard_test/   
+
+if [ "$(cat /tmp/core_sdcard_test/etc/overlayroot.conf | grep "overlayroot=\"tmpfs")" = ""  ];
+then
+        echo "$pf: sdcard tmpfs not properly set up FAIL"
+        exit 3
+fi
+
+chroot /tmp/core_sdcard_test/ ls
+
+trap "if mountpoint -q "/tmp/core_sdcard_test"; then umount /tmp/core_sdcard_test; fi" EXIT
```

Comment:
This trap should be higher in the file ( i usually put all traps at the top).  As if you execute line 26 (the mount) and then exit on line 31) your trap will not be in memory and not execute.  Your `trap` should _always_ be above the code the activates its relevance.